cubic-server/Client.cpp

```diff
@@ -293,6 +293,18 @@ void Client::_onEncryptionResponse(const std::shared_ptr<protocol::EncryptionRes
     _log->debug("Got a Encryption Response");
 }
 
+void Client::_onDisconnectRequest(const std::shared_ptr<protocol::DisconnectReason> &pck, chat::Message &message)
```

Comment:
This function is great but you don't need to use
```cpp
const std::shared_ptr<protocol::DisconnectReason> &pck
```
and there isn't any request of disconnection. Maybe you should name your function like this :
```cpp
void Client::DisconnectClient(chat::Message &message)
```